scripts/rundocker.sh

```diff
@@ -7,21 +7,38 @@ CONTAINER_HOME=/home/kivy
 # Specifies the name of the docker volume used to store p4a cache
 P4A_CACHE=p4a_cached_dir_$ARCH
 
+# Docker volume to cache gradlew download
+GRADLEW_CACHE=gradlew_cached_dir
+GRADLEW_PATH=${CONTAINER_HOME}/.gradle/wrapper
+
 CID_FILE=kolibri-android-app-container-id.cid.txt
 
+CONTAINER_ID=
+cleanup() {
+  set +e
+  if [ -n "${CONTAINER_ID}" ]; then
+    echo -e "Stopping ${CONTAINER_ID}"
+    docker stop "${CONTAINER_ID}"
+    echo -e "Removing ${CONTAINER_ID}"
+    docker rm "${CONTAINER_ID}"
+  fi
+}
+trap cleanup EXIT
+
 # create the container to be used throughout the script
 # creates a volume for reuse between builds, holding p4a's android distro
 docker create -it \
   --mount type=volume,src=${P4A_CACHE},dst=${CONTAINER_HOME}/.local \
+  --mount type=volume,src=${GRADLEW_CACHE},dst=${GRADLEW_PATH} \
```

Comment:
https://github.com/learningequality/kolibri-installer-android/pull/107/commits/63010a261e677e9872a5a09b37ef5dbcd8a325e7
> Without this, gradlew downloads gradle every time the container is run.

🙌

Reply:
I did get really tired of watching that get downloaded.

I think this could actually be the entire `~/.gradle` directory without an issue. Before I figured out the `GRADLE_OPTS` environment variable, I was messing with `~/.gradle/gradle.properties`. Then I had a conflict with managing the volume vs pre-seeding part of it.